File: csharp/hamming/Hamming.cs
using System;
using System.Linq;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length)
        {
            throw new ArgumentException("DNA strands should be of equal length.");
        }

        return firstStrand.ToCharArray().Zip(secondStrand.ToCharArray()).Count(pair => pair.First != pair.Second);
    }
}

File: scala/hamming/src/main/scala/Hamming.scala
object Hamming {
  def distance(first: String, second: String): Option[Int] =
    Option.apply((first.toCharArray, second.toCharArray))
      .filter { case (a, b) => a.length == b.length }
      .map { case (a, b) => a.zip(b) }
      .map(_.count { case (a, b) => a != b })
}